xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp

```diff
@@ -506,6 +506,19 @@ void CDVDVideoCodecDRMPRIME::SetPictureParams(VideoPicture* pVideoPicture)
   pVideoPicture->iWidth = m_pFrame->width;
   pVideoPicture->iHeight = m_pFrame->height;
 
+  /* according to ffmpeg devs crop fields in AVFrame are not always maintained actively
+   * and may cause under-shot buffers or mis-aligned strides if not implemented correctly
+   * in decoders. drm_prime frames on the other hand has their own memory allocators,
+   * and as long as the plane number is single, there should be no future regression
+   * related to usage of crop fields.
+   */
+  AVDRMFrameDescriptor* desc = reinterpret_cast<AVDRMFrameDescriptor*>(m_pFrame->data[0]);
+  if (m_pFrame->format == AV_PIX_FMT_DRM_PRIME && desc->nb_layers == 1)
```

Comment:
Did a quick look at `malidp` driver in mainline, it also seem to signal afbc cropping in similar way, so should probably be okay.

Maybe only try to set the offsets when an ARM frame buffer is used and possible also reset offsets to 0 in an else?

Reply:
> The comment mention "plane number is single" however this code checks for layers, something that typically always will be 1.

Ah yes, good catch, i had implemented this check to prevent exactly the cases like this, but it seems i have mixed `nb_planes` vs `nb_layers`. I think this itself should prevent the issue from happening.
The rationale here is, the offsets should be there in case there is a form of compression, because `generally` whenever there is a form of compression, or specific modifier it is generally possible that there are offset due to frame modifier headers (not a norm but defacto experience).

May be it is even a better approach to check the frame modifier is `LINEAR` or not directly, because even though it is not the practical case, it is still possible to provide multiple planes with modifiers, so there would be actual offsets. So checking the modifier is more universal. Still it is possible that there might be offsets due to alignment requirements with LINEAR mode, but i have never seen such a frame yet.

> I am guessing this offset handling is added to assist with following MPP specific AFBC header offset?

Yes and sometimes due to alignment requirements for compressed frames.

> Do we have mainline Linux kernel code that use similar offset to skip the AFBC header, or is this offset only used in downstream/vendor kernel and may need more review and be reworked to be upstream ready?

Tbh, i do not know, but i can not imagine any other way? The thing is AFBC video decoding is a foreign concept is mainline to my knowledge. There is no V4L based decoder supports it afaik.

https://drmdb.emersion.fr/properties/4008636142/SRC_X

Reply:
@Kwiboo fixed in https://github.com/xbmc/xbmc/pull/24431/commits/1ecb17125a353bbd914c423f114f16c2ce889c01

i found out best way to prevent the issue is not to do heuristic checks, which is cleaner, and tbh, right way to fix such issues is supposed to be ffmpeg rather than here. And from my understanding there is ffmpeg issue at all.